lib/availability_validator.rb

```diff
@@ -0,0 +1,12 @@
+class AvailabilityValidator
```

Comment:
I can see what the intent behind this class was but it feels like, if its only purpose is to check the state of Board, that this might be a bit of "feature envy" and Board could just handle this on its own. Was there something that prompted making this a separate class?

Reply:
Good question! I made it a separate class based on having a number validator, but I see what you mean by it having "feature envy" since its purpose is so related to the state of Board. I'll work on moving that functionality into Board.